examples/tutorial_dhnx_pandapipes/tutorial_dhnx_pandapipes.py

```diff
@@ -0,0 +1,788 @@
+# Note that we only model the forward pipeline system in this example and
+# focus on the pressure losses due to the pipes (no pressure losses e.g. due
+# to expansion bend and so on).
+# However, if we assume the same pressure drop for the return pipes and add
+# a constant value for the substation, we can a first idea of the hydraulic
+# feasibility of the drafted piping system, and we can check, if the
+# temperature at the consumers is sufficiently high.
+
+pp_net = pp.create_empty_network(fluid="water")
+
+for index, fork in forks.iterrows():
+    pp.create_junction(
+        pp_net, pn_bar=pressure_pn, tfluid_k=feed_temp,
+        name=fork['id_full']
+    )
+
+for index, consumer in consumers.iterrows():
+    pp.create_junction(
+        pp_net, pn_bar=pressure_pn, tfluid_k=feed_temp,
+        name=consumer['id_full']
+    )
+
+for index, producer in producers.iterrows():
+    pp.create_junction(
+        pp_net, pn_bar=pressure_pn, tfluid_k=feed_temp,
+        name=producer['id_full']
+    )
+
+# create sink for consumers
+for index, consumer in consumers.iterrows():
+    pp.create_sink(
+        pp_net,
+        junction=pp_net.junction.index[
+            pp_net.junction['name'] == consumer['id_full']][0],
+        mdot_kg_per_s=consumer['massflow'],
+        name=consumer['id_full']
+    )
+
+# create source for producers
+for index, producer in producers.iterrows():
+    pp.create_source(
+        pp_net,
+        junction=pp_net.junction.index[
+            pp_net.junction['name'] == producer['id_full']][0],
+        mdot_kg_per_s=consumers['massflow'].sum(),
+        name=producer['id_full']
+    )
+
+# EXTRENAL GRID as slip (Schlupf)
+for index, producer in producers.iterrows():
+    pp.create_ext_grid(
+        pp_net,
+        junction=pp_net.junction.index[
+            pp_net.junction['name'] == producer['id_full']][0],
+        p_bar=pressure_net,
+        t_k=feed_temp,
+        name=producer['id_full'],
+    )
+
+# create pipes
+for index, pipe in pipes.iterrows():
+    pp.create_pipe_from_parameters(
+        pp_net,
+        from_junction=pp_net.junction.index[
+            pp_net.junction['name'] == pipe['from_node']][0],
+        to_junction=pp_net.junction.index[
+            pp_net.junction['name'] == pipe['to_node']][0],
+        length_km=pipe['length'] / 1000,  # convert to km
+        diameter_m=pipe["Inner diameter [m]"],
+        k_mm=pipe["Roughness [mm]"],
+        alpha_w_per_m2k=pipe["alpha [W/m2K]"],
+        text_k=ext_temp,
+        name=pipe['id'],
+    )
+
+# # Execute the pandapipes simulation
+
+pp.pipeflow(
+    pp_net, stop_condition="tol", iter=3, friction_model="colebrook",
+    mode="all", transient=False, nonlinear_method="automatic", tol_p=1e-3,
+    tol_v=1e-3,
+)
+
+print(pp_net.res_junction.head(n=8))
+print(pp_net.res_pipe.head(n=8))
+
+# # Example of exports of the results
+
+# to excel
+with pd.ExcelWriter('results/results_fine.xlsx') as writer:
```

Comment:
```suggestion
filepath = 'results/results_fine.xlsx'
if not os.path.exists(os.path.dirname(filepath)):
    os.makedirs(os.path.dirname(filepath))
with pd.ExcelWriter(filepath) as writer:
```
The 'results' folder does not exist when the script runs for the first time, so we need to create it here